### ISSUE TITLE
Single file mode breaks when cache is enabled and URL has percent-encoding

### ISSUE THREAD

**Issue by Tubeee** [fangfufu/httpdirfs#128]
To reproduce: take any URL and append `?%21`, e.g. 
```sh
httpdirfs --cache --single-file-mode 'https://cdimage.debian.org/debian-cd/current/amd64/iso-cd/debian-12.2.0-amd64-netinst.iso?%21' mnt
```
which spits out `cannot open 'mnt/debian-12.2.0-amd64-netinst.iso?%21' for reading: No such file or directory` when attempting to read.

To clarify, this only happens with cache enabled, otherwise it works normally.

**Comment by fangfufu** [comment]
What if the URL isn't percent-encoded, and the cache mode is enabled, does the single-file mode still work?

**Comment by fangfufu** [comment]
Thanks for the extra information. I think the cache system follows a different name decoding path. I will get this fixed eventually. I am just a bit occupied with my personal life recently.

**Comment by fangfufu** [comment]
Fixed via 2fa2bb9cf068221b0fb625e600af8fe33f181214
